src/components/skills-quiz/SearchProgramCard.jsx

```diff
@@ -0,0 +1,295 @@
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import {
+  Badge, Card, Icon, StatusAlert,
+} from '@edx/paragon';
+import { Program } from '@edx/paragon/icons';
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+import { SearchContext } from '@edx/frontend-enterprise-catalog-search';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearchMinus } from '@fortawesome/free-solid-svg-icons';
+import { SkillsContext } from './SkillsContextProvider';
+import { isDefinedAndNotNull } from '../../utils/common';
+import { ELLIPSIS_STR } from '../course/data/constants';
+import { shortenString } from '../course/data/utils';
+import { SKILL_NAME_CUTOFF_LIMIT, MAX_VISIBLE_SKILLS_PROGRAM, NO_PROGRAMS_ALERT_MESSAGE } from './constants';
+import getCommonSkills from './data/utils';
+import { useSelectedSkillsAndJobSkills } from './data/hooks';
+import { useDefaultSearchFilters } from '../search/data/hooks';
+import { ProgramType } from '../search/SearchProgramCard';
+import { UserSubsidyContext } from '../enterprise-user-subsidy';
+
+const linkToProgram = (program, slug, enterpriseUUID, programUuid) => {
+  if (!Object.keys(program).length) {
+    return '#';
+  }
+  const { userId } = getAuthenticatedUser();
+  sendEnterpriseTrackEvent(
+    enterpriseUUID,
+    'edx.ui.enterprise.learner_portal.search.program.card.clicked',
+    {
+      userId,
+      programUuid,
+    },
+  );
+  return `/${slug}/program/${programUuid}`;
+};
+
+const renderDialog = () => (
+  <div className="lead d-flex align-items-center py-3">
+    <div className="mr-3">
+      <FontAwesomeIcon icon={faSearchMinus} size="2x" />
+    </div>
+    <p>
+      { NO_PROGRAMS_ALERT_MESSAGE }
+    </p>
+  </div>
+);
+
+const SearchProgramCard = ({ index }) => {
+  const { enterpriseConfig } = useContext(AppContext);
+  const { slug, uuid } = enterpriseConfig;
+  const { subscriptionPlan, subscriptionLicense, offers: { offers } } = useContext(UserSubsidyContext);
+  const offerCatalogs = offers.map((offer) => offer.catalog);
+  const { filters } = useDefaultSearchFilters({
+    enterpriseConfig,
+    subscriptionPlan,
+    subscriptionLicense,
+    offerCatalogs,
+  });
+  const { state } = useContext(SkillsContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const [programs, setPrograms] = useState([]);
+  const [hitCount, setHitCount] = useState(undefined);
+  const { refinements } = useContext(SearchContext);
+  const { skill_names: skills } = refinements;
+  const { selectedJob } = state;
+  // Top 3 recommended programs are determined based on job-skills only, coming either from Search Job or Current Job
+  const selectedJobSkills = useSelectedSkillsAndJobSkills({ getAllSkills: false });
+  const skillsFacetFilter = useMemo(
+    () => {
+      if (selectedJobSkills) {
+        return selectedJobSkills.map((skill) => `skill_names:${skill}`);
+      }
+      return [];
+    },
+    [selectedJob],
```

Comment:
Shouldn't this be `selectedJobSkills`?

Reply:
We want to fetch the programs whenever a learner moves b/w jobs, that's why I have added `selectedJob` here